pandas/core/internals/managers.py

```diff
@@ -390,6 +398,14 @@ def putmask(self, mask, new, align: bool = True):
             align_keys = ["mask"]
             new = extract_array(new, extract_numpy=True)
 
+        if isinstance(mask, ABCDataFrame) and mask._mgr.any_extension_types:
```

Comment:
is there a viable option to avoid getting here with DataFrames?

Reply:
I suspect something could be done in `NDFrame._where` to avoid most cases. And then maybe `.clip` could just call `_where`?

If you're already thinking about changes to `._where` in #51547 I can close this if it makes sense to combine the two.

Reply:
I've moved this to `NDFrame._where`.